### ISSUE TITLE
Update blacklisting is unintuitve

### ISSUE THREAD

**Issue by tabepic** [cromnix/pamac-classic#16]
Basically what is said here: https://github.com/manjaro/pamac/issues/371

I had no idea that this was going on, and personally find it pointless to blacklist them (why not just enter it in the preferences if you want to blacklist it?).
I think it should work as @notramo suggested in that thread:
> What do you think about auto-deselecting in every update, until it is selected again, but not blacklisting?

**Comment by notramo** [comment]
The only problem with my idea is that when there aren't more updates than deselected ones, what should Pamac do? Open a window with full of deselected updates? Or say that system is up to date? Some solution should be figured out.

**Comment by tabepic** [comment]
I think deselected updates, because they still aren't done. Saying that it's up to date it just plain wrong, in my opinion.

**Comment by notramo** [comment]
OK, but what should it do instead?

**Comment by tabepic** [comment]
What do you mean? (I updated my previous comment for clarity)

**Comment by notramo** [comment]
Oh, sorry I misunderstood your previous comment. I agree.

**Comment by cromerc** [comment]
I can't reproduce what he is talking about in neither pamac nor pamac-classic. When I deselect a package from the update tab then only do some updates. The package I deselected is not "blacklisted". It is still in the list of packages that need updating and pamac is still showing that it needs to be updated... 

I also had a look through the code in transaction.vala, there is absolutely no code that adds those packages to the blacklist used in pacman.conf during updates. As far as I can tell, that issue is bogus.

And packages that are truly blacklisted, those in pacman.conf don't show up in pamac's update tab at all, this is what is expected, they should not be shown since they are not going to be updated.

**Comment by tabepic** [comment]
I don't know what to say...
![update discrepancy](https://user-images.githubusercontent.com/9892196/33248171-bd7ef554-d2e0-11e7-8b72-aba817c2f9e8.png)
This is right after updating databases (in both pacman and pamac-classic)

Preferences from pamac-classic:
![prefs](https://user-images.githubusercontent.com/9892196/33248214-10c333b0-d2e1-11e7-81dd-d9e6619466b4.png)

**Comment by cromerc** [comment]
I had something similar to your photo happen to me before, it was caused by what I believe was an out of sync mirror.

**Comment by cromerc** [comment]
If it happens again, could you check the contents of /tmp/pamac-checkdbs/sync and let me know what is there?

**Comment by tabepic** [comment]
Definitely, will do. (I might have a second box with this as well)

**Comment by tabepic** [comment]
Okay, second box has this issue (pacman -Su shows 44 while pamac-classic shows 40).

I noticed a couple of things, one that the first user to use pamac (I assume) has ownership on /tmp/pamac-checkdbs, second being that there was a db.lck file in /tmp/pamac-checkdbs, permissions 0000, owned by the same user.

Attached is a (sadly) ZIP file, because GitHub doesn't like TAR:
[sync.zip](https://github.com/cromnix/pamac-classic/files/1504658/sync.zip)

**Comment by cromerc** [comment]
The lock file is why the updates were not showing up from before. The lock prevented pamac from checking for new updates. The lock is removed on reboot since /tmp gets erased.

As far as the difference between number of packages, I have discovered that pacman and pamac do not show the information in the same way.

pacman shows all packages that will be effected.
pamac shows only the packages that have upgrades available.

Right now on my computer pacman is showing "52" updates and pamac is saying "50".
The difference is because 1 package is going to be removed during the upgrade process(gtk3-print-backends-3.22.26-1), and the other package is going to be newly installed(libcloudproviders-0.2.5-2) because I don't currently have that package installed on my computer.
So to summarise my current upgrade situation:
To upgrade: 50
To remove: 1
To install: 1

pacman just lumps all of those 3 things together and says 52 packages will be effected.

During the actual upgrade via pamac it does remove the no longer needed package and installs the new dependency. So in the end upgrading via pamac and upgrading via pacman is the same, what is different is the information shown to the user in pamac compared to what is shown in pacman.

**Comment by cromerc** [comment]
I m going to close this for now, if you have the same issues please feel free to comment or reopen the issue.
